packages/router/addon/index.js

```diff
@@ -12,6 +14,12 @@ function lazyBundle(routeName) {
 }
 
 export default EmberRouter.extend({
+  init(...args) {
+    this._super(...args);
+    this._inFlightLazyRoutes = 0;
+    registerWaiter(this._doneLoadingLazyRoutes.bind(this));
```

Comment:
At the least you need to guard this with an `if (DEBUG)` flag, otherwise prod builds will be borked.

Also, registering in `init` and not unregistering in `willDestroy` will leave tests in a pretty weird /bad state.

Reply:
See where we setup `Ember.Test.registerWaiter` here: 

https://github.com/emberjs/ember.js/blob/v3.10.1/packages/ember/index.js#L641-L650

Note, there is no `Ember.Test` in production builds.